### ISSUE TITLE
RISubgraphFinder has problems with directed graphs

### ISSUE THREAD

**Issue by cleberecht** [singa-bio/singa#82]
Trying to find
pattern:
![target](https://cdn.pbrd.co/images/HACq5Wz.png)
in target:
![pattern](https://cdn.pbrd.co/images/HACqjJ7.png)
results in

```
Exception in thread "main" java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.util.ArrayList.rangeCheck(ArrayList.java:657)
	at java.util.ArrayList.get(ArrayList.java:433)
	at bio.singa.mathematics.algorithms.graphs.isomorphism.RISubgraphFinder.growSearchSpace(RISubgraphFinder.java:235)
	at bio.singa.mathematics.algorithms.graphs.isomorphism.RISubgraphFinder.growSearchSpace(RISubgraphFinder.java:313)
	at bio.singa.mathematics.algorithms.graphs.isomorphism.RISubgraphFinder.matchTargetGraph(RISubgraphFinder.java:194)
	at bio.singa.mathematics.algorithms.graphs.isomorphism.RISubgraphFinder.<init>(RISubgraphFinder.java:61)
	at bio.singa.mathematics.algorithms.graphs.isomorphism.RISubgraphFinder.<init>(RISubgraphFinder.java:45)
	at bio.singa.javafx.renderer.graphs.DirectedGraphPlayground.main(DirectedGraphPlayground.java:57)
```

In branch `feature/directed-graph-drawing` i have created a test `RISubgraphFinderTest` and a playground `DirectedGraphPlayground` for this issue.

**Comment by fkaiserbio** [comment]
- should be fixed in be54129
- before closing this issue, someone should double check

**Comment by fkaiserbio** [comment]
tested for cyclic directed graphs, closing now
